coalib/results/Diff.py

```diff
@@ -70,6 +72,57 @@ def from_string_arrays(cls, file_array_1, file_array_2, rename=False):
         return result
 
     @classmethod
+    def from_unified_diff(cls, unified_diff, original_file):
+        """
+        Creates a ``Diff`` object from given unified diff.
+
+        If this ``Diff`` is applied to the original file, the corrected
+        file will be created.
+
+        :param unified_diff:  Unified diff string.
+        :param original_file: The contents of the original file
+                              (line-splitted).
+        """
+        patch_set = PatchSet(unified_diff.splitlines())
+
+        diff = Diff(original_file)
+
+        for file in patch_set:
+            for hunk in file:
+                file_line = hunk.source_start
+                i = 0
+                while i < len(hunk):
+                    line = hunk[i]
+                    source_code = str(line)[1:]
+
+                    if line.is_added:
+                        add_set = []
+                        # Add all the consecutive '+' lines at once
```

Comment:
you should describe why you do this shortly: `Diff` doesn't allow to make changes to a place more than once :+1: